addons/radar/functions/fnc_getIcon.sqf

```diff
@@ -63,8 +63,17 @@ if (getText(configFile >> "CfgWeapons" >> (secondaryWeapon (_unit)) >> "UIPictur
 };
 
 // Medic
-if (_unit getVariable ["ace_medical_medicClass", [0, 1] select (_unit getUnitTrait "medic")] > 0) exitWith {
-    _namespace getVariable ["medic", DUI_MEDIC];
+if (GVAR(ace_medic)) then {
+
+    if (_unit getVariable ["ace_medical_medicClass", 0] > 0) exitWith {
```

Comment:
This check should still be the original check, ACE does not always set their variable and has also a fall back to the vanilla trait.

Other than that, SQF works with scopes, exitWith exits current scope. Since you are now inside an sub scope with `if (GVAR(ace_medic)) then {`, you will exit out of the if scope, and the rest of the function will still run.

I'd say you'd need 2 exitWith in the main scope. Both checking if the newly added CBA setting is set or not. Combine that with lazy evaluation and it should be fine enough.

```sqf
if (GVAR(ace_medic) && {_unit getVariable ["ace_medical_medicClass", [0, 1] select (_unit getUnitTrait "medic")] > 0}) exitWith {
    //...
};
if (!GVAR(ace_medic) && {_unit getUnitTrait "medic"}) exitWith {
    //...
};
```